tools/ts_library_builder/ast_util.ts

```diff
@@ -281,8 +283,17 @@ export function namespaceSourceFile(
     }
   });
 
+  // TODO need to properly unwrap this
   const globalNamespace = sourceFile.getNamespace("global");
-  const globalNamespaceText = globalNamespace && globalNamespace.print();
+  let globalNamespaceText = "";
+  if (globalNamespace) {
+    const structure = globalNamespace.getStructure();
+    if (structure.bodyText && typeof structure.bodyText === "string") {
+      globalNamespaceText = structure.bodyText;
```

Comment:
@kitsonk I'm looking at this PR for ideas for improving the library. Are you getting the body text from the structure because it doesn't contain the leading indentation?

If so, I was previously trying to think of a way to nicely expose that otherwise, but haven't thought of a way yet. I just opened https://github.com/dsherret/ts-simple-ast/issues/470

Reply:
@dsherret actually what I am trying to do, and this is a bit of hackery right now, is transform from a UMD/modular `.d.ts` a global scope augmentation to an ambient/global `.d.ts`.  Ideally I would have iterated through the members and applied the `declare` keyword, but for now, just "unwrapping" it works.  From an indentation perspective, I just run it through Prettier instead of relying on trying to keep things straight in the manipulation.

So I want to take something like this:

**someApi.d.ts**

```ts
global {
  interface Foo {
    foo: string;
  }
}

export const foo: Foo;
export type FooAlias = Foo;
```

into this:

**lib.custom.d.ts**

```ts
declare interface Foo {
  foo: string;
}

declare namespace someApi {
  const foo: Foo;
  type FooAlias = Foo;
}
```

Reply:
Ah, ok thanks for the explanation! I remember now that prettier is used at the end for formatting.

There is actually an `.unwrap()` method on classes like `NamespaceDeclaration` and `FunctionDeclaration`, but I don't think that would be appropriate here if you just want the body's text in order to insert it into a different file later. One alternative for now is `namespaceDeclaration.getChildSyntaxListOrThrow().getFullText()` (In the future, I am thinking to add `namespaceDeclaration.getBodyText()` as that seems more straightforward and there's already a `setBodyText` method... updated that issue). What's done here works, but it's a little verbose.